ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,225 @@
+
+        Returns:
+             list: list containing alert metrics
+        """
+        response = self.alerts()
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        return alert_list
+
+    def get_firing_alerts(self, alert_name):
+        """
+        get all the firing alerts from list of active alerts
+        """
+        alert_list = self.get_all_alerts_by_alert_name(alert_name=alert_name)
+        firing_alerts = []
+        for alert in alert_list:
+            if alert["state"] == "firing":
+                firing_alerts.append(alert)
+
+        return firing_alerts
+
+    def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for an alert if found in fired state
+
+        Args:
+             alert (str): alert name
+             timeout (int): wait time, default is 10 mins
+
+        Return:
+             sample (list): list of all alerts that match the alert name and in firing state
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.get_firing_alerts,
+            alert_name=alert_name,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in firing state.")
+                    return sample
+
+        except TimeoutExpiredError:
+            LOGGER.error(f"{alert_name} currently not in firing state")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int: scrape time interval or default 30 if not found
+        """
+        response = self._get_response(query=f"{self.api_v1}/targets")
+        result = response["data"]["activeTargets"]
+        for item in result:
+            if item and item["labels"]["job"] == "prometheus-k8s":
+                scrape_interval = item["scrapeInterval"]
+                return int((re.match(r"\d+", scrape_interval)).group())
+            else:
+                return 30
+
+    def query_sampler(self, query, timeout=TIMEOUT_2MIN):
+        """
+        Sample output for query function
+
+        Args:
+             query (str): prometheus query string
+             wait_timeout (int): default is 2 mins
+
+        Return:
+             list: return the query result
+
+        Raise:
+             TimeoutExpiredError: if query response doesn't return success
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.query,
+            query=query,
+        )
+        sample = None
```

Comment:
`sample = None` is not needed (Not used outside the forloop)

Reply:
done